functions/defaulted_parameters.md

```diff
@@ -0,0 +1,87 @@
+# Module name: topic name
+_Skeleton instructions are typeset in italic text._
+
+## Overview
+
+Functions in C++ may be overloaded with different numbers and types of parameters. It may be of value to specify default values for some number of parameters, to allow a consumer to avoid specifying parameters that rarely change, or to enable expanding the set of parameters while maintaining backward compatibility with existing consumers.
+
+<table>
+  <thead>
+    <th>Level</th>
+    <th>Objectives</th>
+  </thead>
+  <tr>
+    <td>Foundational knowledge</td>
+    <td>can make calls to functions with or without defaulted parameters</td>
+  </tr>
+  <tr>
+    <td>Advanced</td>
+    <td>declares and defines functions with defaulted parameters</td>
+  </tr>
+  <tr>
+    <td>Further studies</td>
+    <td>refinement of defaulted parameters through multiple declarations</td>
+  </tr>
+</table>
+
+## Motivation
+
+Default parameters allow the omission of parameters with obvious or common values. Also may be utilized to extend an existing function signature without forcing changes to existing calling code.
```

Comment:
This raises an interesting discussion point/caveat: At first I thought you were suggesting that one use-case for default arguments is if some third-party library provides the existing signature `void foo(int x, int y)`, the student might consider overriding it in their own code with `void foo(int x, int y = 42)` to save on typing without breaking ABI.
But then I decided that you must mean that if _the student_ has existing code that does `bool bar(int x)`, they can replace all declarations of that function with `bool bar(int x, int y = 42)` to preserve API... but notice that that change breaks ABI instead! (Also [breaks API for anyone who used `bar` as a function pointer](https://quuxplusone.github.io/blog/2020/04/18/default-function-arguments-are-the-devil/#surprising-function-pointer-type) — e.g. `std::find_if(first, last, bar)` — but that's orthogonal.)

Reply:
From a practical standpoint, I often find default parameters come from having a function, `foo(int x)` where the maintainer needs to add a parameter to become `foo(int x, int y)` but to avoid breaking all existing calls, they give it a default value. I've seen a lot of code bases but never one which actually cared about its ABI compatibility. Can certainly be omitted, because students will invariably re-discover this use if it matters to them, but I included out of my own notion of completeness. Happy to remove if strong opinions exist